src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/steps/base-step/base-step.component.ts

```diff
@@ -0,0 +1,109 @@
+import { BaseComponent } from "src/app/common/base.component";
+import { ControlType, JsonFormData } from "../../add-polling-source-form.types";
+import { RadioControlType } from "../../form-control.source";
+import { FormBuilder } from "@angular/forms";
+import { ControlContainer, FormGroupDirective } from "@angular/forms";
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Input,
+    OnInit,
+} from "@angular/core";
+import { FormGroup } from "@angular/forms";
+import { JsonFormControls } from "../../add-polling-source-form.types";
+import { getValidators } from "src/app/common/data.helpers";
+
+@Component({
+    selector: "app-base-step",
+    templateUrl: "./base-step.component.html",
+    styleUrls: ["./base-step.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    viewProviders: [
+        { provide: ControlContainer, useExisting: FormGroupDirective },
+    ],
+})
+export class BaseStepComponent extends BaseComponent implements OnInit {
+    public parentForm: FormGroup;
+    @Input() public sectionStepRadioData: RadioControlType[];
+    @Input() public sectionFormData: JsonFormData;
+    @Input() public defaultKind: string;
+    @Input() public groupName: string;
+    @Input() public title: string;
+    public controlType: typeof ControlType = ControlType;
+    public readonly KIND_NAME_CONTROL = "kind";
+    public readonly SCHEMA_NAME_CONTROL = "schema";
+    private readonly DEFAULT_EVENT_TIME_SOURCE = "fromMetadata";
+
+    constructor(
+        private rootFormGroupDirective: FormGroupDirective,
+        private fb: FormBuilder,
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.parentForm = this.rootFormGroupDirective.form;
+        this.initForm(
+            this.sectionForm.get(this.KIND_NAME_CONTROL)?.value as string,
+        );
+        this.chooseFetchKind();
+    }
+
+    public get sectionForm(): FormGroup {
+        return this.parentForm.get(this.groupName) as FormGroup;
+    }
+
+    public chooseFetchKind(): void {
```

Comment:
Any reason for this method to be public?

Reply:
Yes, it's better to keep it private.